Source/WebCore/css/MediaQueryList.cpp

```diff
@@ -104,7 +112,24 @@ void MediaQueryList::setMatches(bool newValue)
 
 bool MediaQueryList::matches()
 {
-    if (m_matcher && m_evaluationRound != m_matcher->evaluationRound())
+    if (!m_matcher)
+        return m_matches;
+
+    if (RefPtr document = dynamicDowncast<Document>(scriptExecutionContext())) {
+        if (RefPtr ownerElement = document->ownerElement()) {
+            bool isViewportDependent = false;
+            for (auto& query : m_media->queryVector()) {
+                for (auto& expression : query.expressions())
+                    isViewportDependent = isViewportDependent || expression.isViewportDependent();
+            }
```

Comment:
You can bail out from the the loops on the first viewport-dependent expression. It might be be good to make this a lambda.

Reply:
Oh, duh. That's a good point.

Reply:
Will do this instead:

```cpp
bool isViewportDependent = [&]() {
    for (auto& query : m_media->queryVector()) {
        for (auto& expression : query.expressions()) {
            if (expression.isViewportDependent())
                return true;
        }
    }
    return false;
}();
```